packages/CollapsibleChecklists/stories/examples/Basic.js

```diff
@@ -1,150 +1,56 @@
+    { id: 6, item: "Lord of the Rings", isChecked: false },
+    { id: 7, item: "Matrix", isChecked: false },
+  ],
+  Adventure: [
+    { id: 8, item: "Hook", isChecked: false },
+    { id: 9, item: "Avengers", isChecked: false },
+    { id: 10, item: "Toy Story", isChecked: false },
+  ],
+};
 
-  function handleOnChange(changedItemsArray) {
-    action("handleOnChange")();
-    const newSportsData = sportsData.slice(0);
-    newSportsData.forEach(newSportsDatum => {
-      newSportsDatum.sports.forEach(sport => {
-        sport.teams.forEach(team => {
-          const thisTeamWasChanged =
-            changedItemsArray.filter(changedItem => changedItem.props.foobar === team.name).length > 0;
+function BasicStory() {
+  const [items, setItem] = React.useState(_items);
 
-          if (thisTeamWasChanged) {
-            team.isChecked = !team.isChecked; // eslint-disable-line
-          }
-        });
-      });
+  function handleChange(itemsChanged) {
+    const cloneItems = { ...items };
+    itemsChanged.forEach(itemChanged => {
+      const { kind, id } = itemChanged.props;
+      const i = cloneItems[kind].filter(i => i.id === id)[0];
+      i.isChecked = !i.isChecked;
     });
-
-    setSportsData(newSportsData);
-  }
-
-  function renderTeams(teams) {
-    return teams.map(team => (
-      <CollapsibleChecklists.Item
-        foobar={team.name}
-        isChecked={team.isChecked}
-        isDisabled={team.isDisabled}
-        key={team.name}
-      >
-        {team.name}
-      </CollapsibleChecklists.Item>
-    ));
-  }
-
-  function renderOhioFootballTeams() {
-    if (ohioFootball.teams.length === 0) {
-      return <Spinner />;
-    }
-
-    return renderTeams(ohioFootball.teams);
+    setItem(cloneItems);
   }
 
   return (
-    <CollapsibleChecklists onChange={handleOnChange}>
-      <CollapsibleChecklists.Heading>California Sports Teams</CollapsibleChecklists.Heading>
-      <CollapsibleChecklists.Group title={californiaBasketball.title}>
-        {renderTeams(californiaBasketball.teams)}
-      </CollapsibleChecklists.Group>
-      <CollapsibleChecklists.Group title={californiaHockey.title}>
-        {renderTeams(californiaHockey.teams)}
-      </CollapsibleChecklists.Group>
-      <CollapsibleChecklists.Group title={californiaBaseball.title} isDisabled>
-        {renderTeams(californiaBaseball.teams)}
-      </CollapsibleChecklists.Group>
-
-      <CollapsibleChecklists.Heading>Ohio Sports Teams</CollapsibleChecklists.Heading>
-      <CollapsibleChecklists.Group title={ohioFootball.title} onExpand={handleExpand}>
-        {renderOhioFootballTeams()}
-      </CollapsibleChecklists.Group>
-
-      <CollapsibleChecklists.Heading>Saskatchewan Sports Teams</CollapsibleChecklists.Heading>
-      <div style={{ fontStyle: "italic", padding: "4px" }}>None available (this shows that any child is accepted)</div>
+    <CollapsibleChecklists onChange={handleChange}>
+      <CollapsibleChecklists.Heading>Movies</CollapsibleChecklists.Heading>
```

Comment:
@nahumzs Compare this with my snippet: https://github.com/acl-services/paprika/pull/123#issuecomment-512918934
So for example, in addition to "Movies" there would be another "TV Shows" section, "Podcasts" etc. that are in an array. Is that too far-fetched of an example? It's hard to tell now (doesn't look like this will be needed for the current feature). @jamiek-acl already has a similar usage in the advanced story with those sections hard-coded. But as soon as we have it in an array, the current API won't work so we'll need to redesign it and force a breaking change. This is what I'm trying to avoid. Let's chat offline if there is any other misunderstanding

Reply:
Here's another example that doesn't require an outer loop:
```jsx
const Frameworks = () => (
  <React.Fragment>
    <CollapsibleChecklists.Heading />
    {groups} {/* this can be whatever */}
  </React.Fragment>
);

const Projects = () => (
  <React.Fragment>
    {/* similar as above */}
  </React.Fragment>
);

<CollapsibleChecklists>
  <Frameworks />
  <Projects />
</CollapsibleChecklists>
```

This is a totally reasonable way to organize code but will not work either.